processing/src/main/java/org/apache/druid/segment/ObjectBasedColumnSelector.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment;
+
+import org.apache.druid.error.DruidException;
+
+/**
+ * Object based column selector.
+ *
+ * This abstract class provides a {@link ColumnValueSelector} based on the methods of {@link BaseObjectColumnValueSelector}.
+ */
+public abstract class ObjectBasedColumnSelector<T> implements ColumnValueSelector<T>
```

Comment:
I think we need a better name for this because this isn't strictly an object selector like `ObjectColumnSelector` is, and instead since it also implements `BaseLongColumnValueSelector`, `BaseDoubleColumnValueSelector`,
`BaseFloatColumnValueSelector` and `BaseNullableColumnValueSelector`, so its sort of a multi-type selector. It is also important to call the contract of this class - that it doesn't do any real type coercion and only satisfies the numeric selectors if the value is already a `Number`.